src/main/java/org/radarcns/webapp/SubjectEndPoint.java

```diff
@@ -80,6 +83,7 @@ public Response getAllSubjectsJson(
             @PathParam(STUDY_ID) String study
     ) {
         try {
+            checkPermission(getJWT(request), SUBJECT_READ);
```

Comment:
Here we should probably use the `checkPermissionOnProject(getJWT(request), SUBJECT_READ, study)` form. Same for the avro endpoint.

Reply:
This end point will not be used in the future. Instead, The endpoint `/api/mp/subjects` will be used to fetch subject information. Here the studyId is later hardcoded as 0. So we cannot put a permission constraint based on the studyId else it will always give an error. We are keeping this end point active for now but later the subject info will only be fetched from MP